doc/debugging.md

```diff
@@ -127,3 +127,178 @@ default when an ELF has been loaded in CloudpilotEmu.
+a bit more convenient. In order to do so, install the [C/C++ extension for
+VSCode](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools)
+it will provide the necessary integration with GDB.
+
+### Setting up the build task
+
+In order to build the app to be debugged, VSCode needs to be told how
+to invoke the build system. This is done by adding a `tasks.json` file to the
+`.vscode` directory in the project root. Additonally, the build task must
+launch CloudpilotEmu Native with the correct command line options.
+You will need to adjust the following example to match your environment and
+needs, but it should be a good starting point:
+
+```json
+{
+    "version": "2.0.0",
+    "tasks": [
+        {
+            "label": "Elf",
+            "command": "make code0001.68k.elf",
+            "type": "shell",
+            "options": {
+                "cwd": "${workspaceFolder}"
+            },
+            "hide": true
+        },
+        {
+            "label": "Prc",
+            "command": "make",
+            "type": "shell",
+            "options": {
+                "cwd": "${workspaceFolder}"
+            },
+            "hide": true
+        },
+        {
+            "label": "CP",
+            "command": "/path/to/cloudpilotnative -l 6667 path-to-palm-session.bin -s cloudpilotStartDebug.cp",
+            "type": "shell",
+            "options": {
+                "cwd": "${workspaceFolder}"
+            },
+            "hide": true
+        },
+        {
+            "label": "Run",
+            "dependsOn": ["Elf", "Prc", "CP"],
+            "dependsOrder": "sequence",
+            "icon": {
+                "dark": "icon/Run.svg",
+                "light": "icon/Run.svg"
+            },
+            "problemMatcher": {
+                "owner": "cpp",
+                "fileLocation": ["relative", "${workspaceFolder}"],
+                "pattern": {
+                    "regexp": "^(.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
+                    "file": 1,
+                    "line": 2,
+                    "column": 3,
+                    "severity": 4,
+                    "message": 5
+                }
+            }
+        }
+    ]
+}
+```
+
+The example above is composed of several tasks that are chained together. The
+first two tasks are used to build the app. The `CP` task is used to launch
+CloudpilotEmu Native with the correct command line options, and the `Run` task
+chains all the previous tasks together and launches Cloudpilot Native.
+The `Elf` task is optional in the way that it can be removed if the app
+preserves the ELF file during build.
+
+As mentioned above, you will need to adjust the paths and command line options
+to your needs. In particular, you will need to adjust the path to the
+CloudpilotEmu Native executable and the path to the session image.
+
+The session image is the file that CloudpilotEmu Native will load on startup
```

Comment:
Is having a previous version of this app really required for your workflow? This would be weird, as the script below installs the app before setting up the ELF file.

Reply:
As discussed, not needed indeed! Thanks for pointing it out!